src/backend/aspen/api/utils/sample.py

```diff
@@ -273,9 +270,13 @@ def sample_info_to_gisaid_rows(
 ) -> List[Dict[str, str]]:
     gisaid_metadata_rows = []
     for sample_info in submission_information:
-        gisaid_location = " / ".join(
-            [sample_info[key] for key in ["region", "country", "division", "location"]]
-        )
+        # Concat location information to a single string in Region / Country / Division / Location format
+        location_strings = []
+        for key in ["region", "country", "division", "location"]:
+            if sample_info[key]:
```

Comment:
will this raise a keyerror?

Reply:
We construct this dictionary as part of the route (in the above utility function `collect_submission_information()`) so these keys will always be present.

Reply:
The `if` test is just to check for a `None` value, since `str.join()` doesn't handle those nicely.